File: findClosestValueInBST.js
const findClosestValueInBST = (tree, target) => {
    return findClosestHelper(tree, target, Infinity)
}

const findClosestHelper = (tree, target, closestVal) => {
    if (!tree) return closestVal

    if (Math.abs(target - tree.value) < Math.abs(target - closestVal)) {
        closestVal = tree.value
    }
    if (tree.value < target) {
        return findClosestHelper(tree.right, target, closestVal)
    } else if (tree.value > target) {
        return findClosestHelper(tree.left, target, closestVal)
    } else {
        return closestVal
    }
}

File: Binary Search Trees/find_closest_value_in_BST.py
def find_closest_value_in_BST(tree, target):
    return find_closest_helper(tree, target, float("inf"))

def find_closest_helper(tree, target, closest):
    if tree is None:
        return closest
    
    if abs(tree.value - target) < abs(closest - target):
        closest = tree.value
    
    if tree.value > target:
        return find_closest_helper(tree.left, target, closest)
    elif tree.value < target:
        return find_closest_helper(tree.right, target, closest)
    else:
        return closest